xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/CommentDeleteAction.java

```diff
@@ -0,0 +1,159 @@
+                    localizePlainOrReturnKey("platform.core.action.commentRemove.noClassnameSpecified"),
+                    ScriptContext.ENGINE_SCOPE);
+        } else if (classId < 0) {
+            getCurrentScriptContext().setAttribute(attributeName,
+                    localizePlainOrReturnKey("platform.core.action.commentRemove.noCommentSpecified"),
+                    ScriptContext.ENGINE_SCOPE);
+        } else {
+            obj = doc.getObject(className, classId);
+            if (obj == null) {
+                getCurrentScriptContext().setAttribute(attributeName,
+                        localizePlainOrReturnKey("platform.core.action.commentRemove.invalidComment"),
+                        ScriptContext.ENGINE_SCOPE);
+            }
+        }
+
+        return obj;
+    }
+
+    @Override
+    public boolean action(XWikiContext context) throws XWikiException
+    {
+        // CSRF prevention
+        if (!csrfTokenCheck(context)) {
+            return false;
+        }
+
+        XWiki xwiki = context.getWiki();
+        XWikiResponse response = context.getResponse();
+        XWikiDocument doc = context.getDoc();
+        DocumentReference userReference = context.getUserReference();
+
+        // We need to clone this document first, since a cached storage would return the same object for the
+        // following requests, so concurrent request might get a partially modified object, or worse, if an error
+        // occurs during the save, the cached object will not reflect the actual document at all.
+        doc = doc.clone();
+
+        BaseObject obj = getObject(doc, context);
+        if (obj == null) {
+            return true;
+        }
+
+        doc.removeObject(obj);
+        doc.setAuthorReference(userReference);
+
+        String changeComment = localizePlainOrReturnKey("core.comment.deleteComment");
+
+        // Make sure the user is allowed to make this modification
+        context.getWiki().checkSavingDocument(userReference, doc, changeComment, true, context);
+
+        xwiki.saveDocument(doc, changeComment, true, context);
+
+        if (Utils.isAjaxRequest(context)) {
+            response.setStatus(HttpServletResponse.SC_NO_CONTENT);
+            response.setContentLength(0);
+        } else {
+            // forward to edit
+            String redirect = Utils.getRedirect("edit", context);
+            sendRedirect(response, redirect);
+        }
+        return false;
+    }
+
+    @Override
+    public String render(XWikiContext context) throws XWikiException
+    {
+        if (Utils.isAjaxRequest(context)) {
+            XWikiResponse response = context.getResponse();
+            response.setStatus(HttpServletResponse.SC_CONFLICT);
+            response.setContentType("text/plain");
+            try {
+                response.getWriter().write("failed");
+                response.setContentLength(6);
+            } catch (IOException e) {
```

Comment:
Nothing to handle the exception?

Reply:
Note that most parts are the same as `ObjectRemoveAction`. This empty exception handle has been in this other file since 2008 so it seems like it's not a huge problem.
This is bad quality I agree.
I added a log similar to what was done in XWikiAction.java
https://github.com/xwiki/xwiki-platform/blob/4e2efc811fd359eaf87f1bc1e43e78d54777a801/xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/XWikiAction.java#L385 

Addressed in https://github.com/xwiki/xwiki-platform/pull/2836/commits/767daaca4e1b6aa5e143e872c725c742e83c8bbb :+1: